lib/activerecord-import/import.rb

```diff
@@ -961,8 +961,9 @@ def find_associated_objects_for_import(associated_objects_by_class, model)
         changed_objects.each do |child|
           child.public_send("#{association_reflection.foreign_key}=", model.id)
           # For polymorphic associations
+
           association_reflection.type.try do |type|
-            child.public_send("#{type}=", model.class.base_class.name)
+            child.public_send("#{type}=", model.class&.polymorphic_name || model.class.base_class.name)
```

Comment:
I think you might need to use `respond_to?` to determine what call to make.

Reply:
oh great catch! Thank you